.github/workflows/assign-class-label.yaml

```diff
@@ -0,0 +1,41 @@
+name: assign-class-label tests and linting/formatting
+
+on:
+  push:
+    paths:
+    - container-images/assign-class-label/**
+  pull_request:
+    paths:
+    - container-images/assign-class-label/**
+
+jobs:
+  lint-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'  
+
+      - name: Install dependencies
+        working-directory: ./container-images/assign-class-label/
+        run: |
+          pip install -r requirements.txt
+          pip install -r tests/requirements.txt
+
+      - name: Run ruff format (formatting)
+        run: |
+          ruff format
+
+      - name: Run ruff check (linting)
+        run: |
+          ruff check --fix
```

Comment:
Note that you are running `ruff` from the top-level directory, rather than from the `container-images/assign-class-label` directory. That means you will potentially be checking files other than in the `assign-class-label` directory.

If that's intentional, you should move this into a different workflow. Otherwise, you should probably add a `working-directory` option to these tasks as well. Alternately, set a default for the whole workflow in [`defaults.run.working-directory`](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#defaultsrunworking-directory).